js/util/navigationUtil.ts

```diff
@@ -10,13 +10,16 @@ export default class NavigationUtil {
     screen_path: string;
     screen_name: string;
   } | null {
-    if (
-      !(this.heapNavRef && this.heapNavRef.state && this.heapNavRef.state.nav)
-    ) {
+    let rootState: any = null;
+    if (this.heapNavRef && this.heapNavRef.state && this.heapNavRef.state.nav) {
```

Comment:
What is the `heapNavRef` exactly? (when is it set, what component/element are you creating a ref on, why do you need the ref). Sorry bout all the q's 😅

Reply:
This is the ref to the NavigationContainer or AppContainer (depending on React Nav version).  We need it to obtain the current state of navigation (route path and name) for screenview props on events, and screenview props on screenviews